addons/crm/models/crm_lead.py

```diff
@@ -297,6 +297,8 @@ def _compute_company_id(self):
                     proposal = lead.user_id.company_id & self.env.companies
                 elif lead.team_id:
                     proposal = lead.team_id.company_id
+                elif lead.partner_id and lead.partner_id.company_id:
```

Comment:
I don't think we want to force a new company based on the customer's one ... shouldn't we keep it void instead ? Generally this means less issues.

Reply:
1) This review contradicts your first one concerning splitting the `if` condition. We cannot apply both reviews, it would break the fix.
2) I would say that it's best to avoid having a void company in most cases as there are validation upon saving the record. Plus, the way the new condition is placed enforced to avoid having unnecessary warning message upon saving.